pkg/db/advisory_locks.go

```diff
@@ -93,7 +101,9 @@ func (f *AdvisoryLockFactory) NewNonBlockingLock(ctx context.Context, id string,
 	}
 
 	log.V(4).Info(fmt.Sprintf("Locked non blocking advisory lock id=%s type=%s - owner=%s", id, lockType, *lock.uuid))
+	f.mutex.Lock()
```

Comment:
we may make the `advisoryLockMap` thread safe, and we also need commit this to the upstream